Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIrMetadataEmitter.kt

```diff
@@ -0,0 +1,333 @@
+        get() = listOfNotNull(
+                Flag.IS_PUBLIC,
+                Flag.Property.IS_DECLARATION,
+                Flag.IS_FINAL,
+                when (kind) {
+                    is PropertyStub.Kind.Val -> null
+                    is PropertyStub.Kind.Var -> Flag.Property.IS_VAR
+                    is PropertyStub.Kind.Constant -> Flag.Property.IS_CONST
+                },
+                when (kind) {
+                    is PropertyStub.Kind.Constant -> null
+                    is PropertyStub.Kind.Val,
+                    is PropertyStub.Kind.Var -> Flag.Property.HAS_GETTER
+                },
+                when (kind) {
+                    is PropertyStub.Kind.Constant -> null
+                    is PropertyStub.Kind.Val -> null
+                    is PropertyStub.Kind.Var -> Flag.Property.HAS_SETTER
+                }
+        ).let { flagsOf(*it.toTypedArray()) }
+
+    val PropertyStub.getterFlags: Flags
+        get() = listOfNotNull(
+                Flag.HAS_ANNOTATIONS.takeIf { annotations.isNotEmpty() },
+                Flag.IS_PUBLIC,
+                Flag.IS_FINAL,
+                Flag.PropertyAccessor.IS_EXTERNAL
+        ).let { flagsOf(*it.toTypedArray()) }
+
+    val PropertyStub.setterFlags: Flags
+        get() = listOfNotNull(
+                Flag.HAS_ANNOTATIONS.takeIf { annotations.isNotEmpty() },
+                Flag.IS_PUBLIC,
+                Flag.IS_FINAL,
+                Flag.PropertyAccessor.IS_EXTERNAL
+        ).let { flagsOf(*it.toTypedArray()) }
+
+    val StubType.flags: Flags
+        get() = listOfNotNull(
+                Flag.Type.IS_NULLABLE.takeIf { nullable }
+        ).let { flagsOf(*it.toTypedArray()) }
+
+    val TypealiasStub.flags: Flags
+        get() = listOfNotNull(
+                Flag.IS_PUBLIC
+        ).let { flagsOf(*it.toTypedArray()) }
+
+    fun AnnotationStub.map(): KmAnnotation {
+        val args = when (this) {
+            AnnotationStub.ObjC.ConsumesReceiver -> TODO()
+            AnnotationStub.ObjC.ReturnsRetained -> TODO()
+            is AnnotationStub.ObjC.Method -> TODO()
+            is AnnotationStub.ObjC.Factory -> TODO()
+            AnnotationStub.ObjC.Consumed -> TODO()
+            is AnnotationStub.ObjC.Constructor -> TODO()
+            is AnnotationStub.ObjC.ExternalClass -> TODO()
+            AnnotationStub.CCall.CString -> TODO()
+            AnnotationStub.CCall.WCString -> TODO()
+            is AnnotationStub.CCall.Symbol ->
+                mapOf("id" to KmAnnotationArgument.StringValue(symbolName))
+            is AnnotationStub.CStruct -> TODO()
+            is AnnotationStub.CNaturalStruct -> TODO()
+            is AnnotationStub.CLength -> TODO()
+            is AnnotationStub.Deprecated -> TODO()
+        }
+        return KmAnnotation(classifier.fqNameSerialized, args)
+    }
+
+    fun StubType.map(): KmType {
+        return when (this) {
+            is ClassifierStubType -> KmType(flags).also { km ->
```

Comment:
KmType is bad name.

Reply:
It's a part of `kotlinx-metadata` library interface.